integration-tests/contract-security-tickets-join.spec.ts

```diff
@@ -0,0 +1,198 @@
+
+// TC-T-018: Create ticket - call input string/address
+// TC-T-019: Create ticket -  string/address
+// TC-T-020: Join ticket - two different ticketers
+// TC-T-021: Join ticket - two different, but similar cty
+
+CONFIGS().forEach(({ lib, rpc, setup, protocol }) => {
+  const Tezos = lib;
+  const mondaynet = protocol === Protocols.ProtoALpha ? test: test.skip;
+
+  describe(`Test contracts using: ${rpc}`, () => {
+    beforeEach(async (done) => {
+      await setup();
+      done();
+    });
+
+    mondaynet('Verify creating ticket is not possible with call input string/address - parameter passed is invalid for type ticket string', async (done) => {
+      try {
+        const op = await Tezos.contract.originate({
+          code: `   {   parameter (ticket string);
+                        storage (option (ticket string));
+                        code
+                          {
+                            CAR; # drop storage, but leave input on stack
+                            PUSH nat 4;
+                            PUSH string "test";
+                            TICKET;
+                            PAIR;
+                            JOIN_TICKETS;
+                            NIL operation;
+                            PAIR;
+                          }
+                    }`,
+          init: 'None',
+        });
+
+        await op.confirmation();
+        expect(op.hash).toBeDefined();
+        expect(op.includedInBlock).toBeLessThan(Number.POSITIVE_INFINITY);
+        const opContract = await op.contract();
+        const opSend = await opContract.methods.default(`(Pair ${opContract.address} \"test\" 1)`).send();
+        await opSend.confirmation()
+      } catch (error: any) {
+        expect(error.message).toContain('Unable to encode parameter');
+      }
+      done();
+    });
+
+    mondaynet('Verify creating ticket is not possible with string/address', async (done) => {
+      try {
+        const op = await Tezos.contract.originate({
+          code: `   {   parameter (option (ticket string));
+                        storage (option (ticket string));
+                        code
+                          {
+                            CAR; # drop storage, but leave input on stack
+                            IF_NONE { FAIL} {};
+                            PUSH nat 4;
+                            PUSH string "test";
+                            TICKET;
+                            PAIR;
+                            JOIN_TICKETS;
+                            NIL operation;
+                            PAIR;
+                          };
+                        }`,
+          init: 'None',
+        });
+
+        await op.confirmation();
+        expect(op.hash).toBeDefined();
+        expect(op.includedInBlock).toBeLessThan(Number.POSITIVE_INFINITY);
+        const opContract = await op.contract();
+        const opSend = await opContract.methods.default(`(Pair ${opContract.address} \"test\" 1)`).send();
```

Comment:
michelson expression (Pair ${opContract.address} \"test\" 1) is not a valid parameter for methods.default

Reply:
the purpose of the test is to verify that tickets cannot be created by cheating with bogus parameters